Added Objetslitt custom catalog

florian-rieder opened the pull request:
Added a new custom catalog for Objetslitt which contains the following features and customizations:

- Added a "Daily item" feature that uses `Rails.cache.fetch` and expires every day at midnight.
- Reorder the simple search result tabs so that the category "textes" will be shown first if available.
- Customized the home page to display a cards menu and the daily item.
- Added a custom page for the "donnees" slug, which displays a simple search form and the daily item.
- Customized the navbar when visiting the "donnees" page to show a different menu specifically for this page.
- Added a mailto icon to the navbar.
- Added logos in the footer.

florian-rieder committed:
Random object

Hello world and basic implementation of the "Object of the day" feature

florian-rieder committed:
Added persistence of daily object using cache

florian-rieder committed:
Daily object cache valid until EOD

florian-rieder committed:
Access data properly using UUIDs

+ fixed CSS
+ Added description field to daily object display

florian-rieder committed:
Added images assets

florian-rieder committed:
Fixed description truncation

florian-rieder committed:
Home cards menu

florian-rieder committed:
Added images

+ Added footer logos
+ Added home menu cards images
+ Added top right nav mailto, next to the help icon

florian-rieder committed:
Home menu transition

+ fixed footer logo height and spacing

florian-rieder committed:
Styles and small fixes

florian-rieder committed:
Optimized daily item queries and caching

florian-rieder committed:
Added simple search to home

+ banner image css
+ visual tweaks

florian-rieder committed:
improved styling and layout

florian-rieder committed:
Override "donnees" page

florian-rieder committed:
Modularization of daily item

in order to use it in multiple views

florian-rieder committed:
Special menu for donnees page

florian-rieder committed:
Reorder simple search results tabs

Prefer showing the "textes" category by default

florian-rieder committed:
Preparations for PR

- increased size of cards images
- used 2 spaces indentation across source code
- small fix to search customization
- styling fixes

florian-rieder committed:
formatted styles

florian-rieder committed:
reorganized simple search controller + view

lucaguindani commented on .gitignore:
Missing empty new line at the end of the file

lucaguindani commented on objetslitt/views/pages/show.html.erb:
Missing empty new line at the end of the file

lucaguindani commented on objetslitt/views/shared/_daily_item.html.erb:
Missing empty new line at the end of the file

lucaguindani commented on objetslitt/views/shared/navbar/_navbar_catalog.erb:
Overriding this view to add a menu for a specific page is risky... catalog could break if we make changes to the navbar and cannot backport them to the custom catalog. Maybe you could add the dropdowns **under** the navbar by overriding shared/_page_header.erb (this view contains a lot less logic than the navbar -> fewer risks of breaking) ?

Why absolute urls and not relative ? Remove all the `https://catima.unil.ch` ?

florian-rieder replied:
I fully understand the reasons not to override this view. Moving the custom menu for the "donnees" page to the page header is a good idea, beyond avoiding the risky override of navbar_catalog, it also helps with the accessibility and navigability of the catalog by keeping the main menu consistent for every page.

The mailto logo has been moved to the new help_menu view, which avoids overriding navbar_catalog. The absolute urls have been converted to relative.

lucaguindani commented on objetslitt/views/simple_searches/_list.html.erb:
Also risky to override, but this is a feature we're planning to add in the future, so we can keep it like that and we will adapt the catalog when the feature is implemented for everyone.

florian-rieder committed:
Changes as requested

florian-rieder committed:
Removed navbar catalog override

To use the new help menu partial instead, to get the same effect without touching the logic in navbar catalog which is risky.

florian-rieder committed:
Fixed donnees menu dropdown display on mobile

+ moved search bar to page header
+ fixed margin errors in daily item

florian-rieder committed:
Added missing empty newline at the end of _help_menu.html.erb

florian-rieder committed:
Handling of objets without a descripfct

Necessary to prevent the catalog from breaking, since the descripfct field is defined as optional.
